File: src/main/java/ca/weblite/jdeploy/app/repositories/ProjectRepositoryInterface.java
package ca.weblite.jdeploy.app.repositories;

import ca.weblite.jdeploy.app.collections.ProjectSet;
import ca.weblite.jdeploy.app.records.Project;

import java.util.UUID;

public interface ProjectRepositoryInterface {
    Project findOneById(UUID id);

    ProjectSet findRecent();

    Project findOnebyPath(String path);
}

File: src/main/java/ca/weblite/jdeploy/app/repositories/ProjectRepositoryInterface.kt
package ca.weblite.jdeploy.app.repositories

import ca.weblite.jdeploy.app.collections.ProjectSet
import ca.weblite.jdeploy.app.exceptions.NotFoundException
import ca.weblite.jdeploy.app.records.Project
import java.util.*

interface ProjectRepositoryInterface {
    @Throws(NotFoundException::class)
    fun findOneById(id: UUID): Project

    fun findRecent(): ProjectSet

    @Throws(NotFoundException::class)
    fun findOnebyPath(path: String): Project

    fun saveOne(project: Project): Project
}
